InformationScripting/src/dataformat/Tuple.cpp

```diff
@@ -28,16 +28,17 @@
 
 namespace InformationScripting {
 
-Tuple::Tuple(std::initializer_list<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(std::initializer_list<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
-Tuple::Tuple(QList<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(QList<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
 QString Tuple::tag() const
 {
+	if (!tag_.isNull()) return tag_;
```

Comment:
I think an assertion is the right thing. This is not a user error, but a programming error in the sense that one should have provided either a tag or some initial values.